apps/api/src/app/integrations/usecases/get-novu-integration/get-novu-integration.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { AuthModule } from '../../../auth/auth.module';
+import { MessageRepository, IntegrationRepository, ExecutionDetailsRepository } from '@novu/dal';
+import { SharedModule } from '../../../shared/shared.module';
+import { IntegrationModule } from '../../integrations.module';
+import { GetNovuIntegration } from './get-novu-integration.usecase';
+import { GetNovuIntegrationCommand } from './get-novu-integration.command';
+import { ChannelTypeEnum } from '../../../../../../../packages/stateless/src/lib/template/template.interface';
+import { EmailProviderIdEnum } from '../../../../../../../libs/shared/src/consts/providers/provider.enum';
+import { endOfMonth, startOfMonth } from 'date-fns';
+
+describe('Get Novu Integration', function () {
+  let getNovuIntegration: GetNovuIntegration;
+  let executionDetailsRepository: ExecutionDetailsRepository;
+  let integrationRepository: IntegrationRepository;
+  let messageRepository: MessageRepository;
+  let session: UserSession;
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [SharedModule, AuthModule, IntegrationModule],
+      providers: [],
+    }).compile();
+
+    session = new UserSession();
+    await session.initialize({
+      noIntegrations: true,
+    });
+    getNovuIntegration = moduleRef.get<GetNovuIntegration>(GetNovuIntegration);
+    executionDetailsRepository = moduleRef.get<ExecutionDetailsRepository>(ExecutionDetailsRepository);
+    integrationRepository = moduleRef.get<IntegrationRepository>(IntegrationRepository);
+    messageRepository = moduleRef.get<MessageRepository>(MessageRepository);
+  });
+
+  it('should return Novu integration', async function () {
+    const result = await getNovuIntegration.execute(
+      GetNovuIntegrationCommand.create({
+        organizationId: session.organization._id,
+        environmentId: session.environment._id,
+        channelType: ChannelTypeEnum.EMAIL,
+      })
+    );
+
+    expect(result).to.have.length(1);
+    expect(result[0].providerId).to.equal(EmailProviderIdEnum.Novu);
+  });
+
+  it('should not return Novu integration for sms', async function () {
+    const result = await getNovuIntegration.execute(
+      GetNovuIntegrationCommand.create({
+        organizationId: session.organization._id,
+        environmentId: session.environment._id,
+        channelType: ChannelTypeEnum.SMS,
+      })
+    );
+
+    expect(result).to.have.length(0);
+  });
+
+  it('should not return Novu integration if notifications are sent', async function () {
+    sinon.stub(executionDetailsRepository, 'count').resolves(1);
+
+    const result = await getNovuIntegration.execute(
+      GetNovuIntegrationCommand.create({
+        organizationId: session.organization._id,
+        environmentId: session.environment._id,
+        channelType: ChannelTypeEnum.EMAIL,
+      })
+    );
+
+    expect(result).to.have.length(0);
+  });
```

Comment:
I'm not sure I understand this test here, how is it different to the one below?

Reply:
I think we should discuss the rules again after have seen your comments...